animation_nodes/nodes/spline/splines_from_edges_utils.pyx

```diff
@@ -0,0 +1,117 @@
+
+        neighbours[neighboursStarts[edges.data[i].v2] + filledSpaces[edges.data[i].v2]] = edges.data[i].v1
+        filledSpaces[edges.data[i].v2] += 1
+
+    # Find the indices of the vertices that are not connected to two vertices.
+    cdef int *nonBipolarVertices = <int*>PyMem_Malloc(verticesAmount * sizeof(int))
+    cdef int nonBipolarVertsCount = 0
+    for i in range(verticesAmount):
+        if neighboursAmounts[i] != 2:
+            nonBipolarVertices[nonBipolarVertsCount] = i
+            nonBipolarVertsCount += 1
+    nonBipolarVertices = <int*>PyMem_Realloc(nonBipolarVertices, nonBipolarVertsCount * sizeof(int))
+
+    # Generate Splines.
+    cdef list splines = []
+    cdef PolySpline spline
+    cdef FloatList splineRadii
+    cdef Vector3DList splineVertices
+    cdef int nonBipolarVertex, currentVertex, nextVertex
+    for i in range(nonBipolarVertsCount):
+        nonBipolarVertex = nonBipolarVertices[i]
+        for j in range(neighboursAmounts[nonBipolarVertex]):
+            nextVertex = neighbours[neighboursStarts[nonBipolarVertex] + j]
+            if (filledSpaces[nextVertex] == neighboursAmounts[nextVertex] or
+                                           neighboursAmounts[nextVertex] != 2):
+                splineVertices = Vector3DList.__new__(Vector3DList, capacity = verticesAmount)
+                splineRadii = FloatList.__new__(FloatList, capacity = verticesAmount)
+
+                splineVertices.append(vertices[nonBipolarVertex])
+                splineRadii.append(radii.get(nonBipolarVertex))
+                splineVertices.append(vertices[nextVertex])
+                splineRadii.append(radii.get(nextVertex))
+                filledSpaces[nonBipolarVertex] -= 1
+                filledSpaces[nextVertex] -= 1
+
+                currentVertex = nonBipolarVertex
+                for _ in range(verticesAmount):
+                    if neighboursAmounts[nextVertex] == 2:
+                        i = currentVertex
+                        currentVertex = nextVertex
+                        if neighbours[neighboursStarts[nextVertex]] == i:
+                            nextVertex = neighbours[neighboursStarts[nextVertex] + 1]
+                        else:
+                            nextVertex = neighbours[neighboursStarts[nextVertex]]
+                        splineVertices.append(vertices[nextVertex])
+                        splineRadii.append(radii.get(nextVertex))
+                        filledSpaces[nextVertex] -= 1
+                    else:
+                        break
+                splines.append(PolySpline.__new__(PolySpline, splineVertices, splineRadii))
+
+    PyMem_Free(nonBipolarVertices)
+    PyMem_Free(neighboursAmounts)
+    PyMem_Free(neighboursStarts)
+    PyMem_Free(filledSpaces)
+    PyMem_Free(neighbours)
+
+    return splines
+
+def splinesFromEdges(Vector3DList vertices, EdgeIndicesList edges, VirtualDoubleList radii,
+                     bint isVertexRadius):
```

Comment:
would prefer a `str` input here for `isVertexRadius` because "how are the radii used?" is not really a yes-no question.
There could be more than two options in theory.